repository/kafka/operator/templates/bootstrap.yaml

```diff
@@ -97,3 +97,25 @@ data:
     # Provided configuration
     ${KAFKA_CONFIGURATION}
     EOF
+
+    {{ if .Params.CUSTOM_SERVER_PROPERTIES_CM_NAME }}
+    KAFKA_CUSTOM_CONFIGURATION_FILE=$(ls -d /custom-configuration/* | head -n 1)
+    cp $KAFKA_CUSTOM_CONFIGURATION_FILE /tmp/custom-configuration
+    echo "Appending custom configuration file to the server.properties..." | xargs -L 1 echo $(date +'[%Y-%m-%d %H:%M:%S,%3N]') $1
+
+    # remove blacklisted properties that users cannot override
+    skip_properties=("broker.id" "listeners" "advertised.listeners" "advertised.host.name" "listener.security.protocol.map" "log.dirs")
+    for property in "${skip_properties[@]}"; do
+    	if grep -q $property /tmp/custom-configuration; then
+      		echo "WARN: cannot override '$property' using custom properties configmap.";
+      		echo "Removing property '$property'"
+      		sed -i "/^$property/d" /tmp/custom-configuration
```

Comment:
Note this will break if a property name is a prefix of another prorperty's name.
Also it might be more user-friendly to fail on an unsupported property rather than remove it.

Reply:
@porridge these are broker properties and no other property would have e.g: `broker.id.morecontent` that's why I added `advertised.listeners` and `advertised.host.name` as separate properties. I can still add `=` suffix to make it clear here. 

I found bringing a whole Kafka cluster down after an unsupported property is added not a good experience. Rather just drop a warning in logs as those are unsupported properties. Users cannot configure them and that would be documented. We might introduce controlled ways to configure some of them later as new features

Reply:
added `=` suffix to all properties names so if users add `broker.id.x=value` that won't be removed

Reply:
@zmalik it might be true that the properties listed at the moment are unique in the prefix sense. But we need to think about the future - I doubt someone adding more properties to the blacklist in a year will think about this caveat :-)

As for bringing a cluster down - how could this happen? Will a rolling restart not respect health checks? IMNSHO the rolling restart should halt once the first broker does not restart cleanly, and this should alert the user to the misconfiguration.

Reply:
yeah, thats true about rolling restart. But with bringing down I meant that `statefulset` is misconfigured and a broker is flapping.
I'm thinking from the support perspective, I prefer a ticket where the user cannot configure a property vs configuring a property resulted in in a flapping broker.